sympy/solvers/ode/ode.py

```diff
@@ -1845,39 +1829,27 @@ def odesimp(ode, eq, func, hint):
     if eq.lhs == func and not eq.rhs.has(func):
         # The solution is already solved
         eq = [eq]
-
-        # special simplification of the rhs
         if hint.startswith("nth_linear_constant_coeff"):
```

Comment:
Do we still need to check the hint here? It would be better not to have simplification that is specific to one hint if possible.

Reply:
yes, because the way collect is done is specifically for the solutions returned by `nth_linear_constant_coeff` due to roots of characterstic equation. 
If we remove this if statement, It will change the form of solutions of other hints too as they already return solution solved for func.
Example:
```python
>>> eq = (x*log(f(x)/x) - 2*x)*Derivative(f(x), x) + f(x)
>>> dsolve(eq)  # keeping this If condition
Eq(f(x), -exp(C1)*LambertW(-x*exp(1 - C1)))

>>> dsolve(eq) #removing If condition
Eq(f(x), -exp(C1)*LambertW(-E*x*exp(-C1)))
```

Reply:
Ideally we would apply the same simplification routines to the solutions from all solvers. For now we can leave this here but ultimately I think it should be removed.

Perhaps the best result would be to use the `simpsol` function from `systems.py`.